### PR TITLE
Do not check for valid links in semantic conventions.

### PR DESCRIPTION
**Pull request by carlosalberto**
Recently IBM changed docs referenced by our semconv, and hence our CI is broken - but there shouldn't be need to check links for it, as they are outdated (and will be removed eventually?).

### CODE REVIEW

**Comment by carlosalberto** [comment]
Please review @open-telemetry/specs-approvers - else, we have to update our links under semconv (which is also a possibility).

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # All documents to be used in spell check.
-ALL_DOCS := $(shell find . -type f -name '*.md' -not -path './.github/*' -not -path './node_modules/*' | sort)
+ALL_DOCS := $(shell find . -type f -name '*.md' -not -path './.github/*' -not -path './node_modules/*' -not -path '*semantic_conventions*' | sort)
```
Comment (reyang) [inline Makefile:2]: This will also remove the folder/path from other checks (e.g. misspell)?

**Comment by joaopgrassi** [comment]
Not sure, I think it was actually useful the checks found the IBM broken links and now we fixed them? (either by removing or finding the "new" place). If we remove the check then we will never know unless someone finds it, or did I misinterpreted the intention here 🤔

**Comment by joaopgrassi** [comment]
Ah I see, this is in the spec repo, I missed that, sorry!. I guess that would make sense as we will probably remove those from here soon.

File: Makefile (hunk 1 above)
Comment (carlosalberto) [reply Makefile:2]: Oh yes - but hopefully we don't ever change them :)

**Review by reyang — APPROVED** [review]

**Pull request merged**